src/Middleware/HealthChecks/src/HealthCheckResponseWriters.cs

```diff
@@ -1,18 +1,28 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Threading.Tasks;
+using System.Text;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
 
 namespace Microsoft.AspNetCore.Diagnostics.HealthChecks
 {
     internal static class HealthCheckResponseWriters
     {
+        private static readonly byte[] DegradedBytes = Encoding.UTF8.GetBytes(HealthStatus.Degraded.ToString());
+        private static readonly byte[] HealthyBytes = Encoding.UTF8.GetBytes(HealthStatus.Healthy.ToString());
+        private static readonly byte[] UnhealthyBytes = Encoding.UTF8.GetBytes(HealthStatus.Unhealthy.ToString());
+
         public static Task WriteMinimalPlaintext(HttpContext httpContext, HealthReport result)
         {
             httpContext.Response.ContentType = "text/plain";
-            return httpContext.Response.WriteAsync(result.Status.ToString());
+            return result.Status switch
+            {
+                HealthStatus.Degraded => httpContext.Response.Body.WriteAsync(DegradedBytes.AsMemory()).AsTask(),
+                HealthStatus.Healthy => httpContext.Response.Body.WriteAsync(HealthyBytes.AsMemory()).AsTask(),
```

Comment:
@Tratcher I'm looking again at the `AsTask()` at the end of this method. Should I use a method like `ValueTaskExtensions.GetAsTask`?
https://github.com/dotnet/aspnetcore/blob/8b30d862de6c9146f466061d51aa3f1414ee2337/src/Shared/ValueTaskExtensions/ValueTaskExtensions.cs#L13-L26

If I remember correctly `WriteAsync` sometimes returns synchronously because it's buffering, But I'm not sure if this behavior was only for libuv or the socket implementation is doing that as well.

Reply:
Interesting, I haven't used those extensions. @BrennanConroy?

Yes WriteAsync can often complete synchronously because it writes into Kestrel's response buffer pipe.